application/src/main/java/org/thingsboard/server/controller/OtaPackageController.java

```diff
@@ -186,38 +182,21 @@ public OtaPackageInfo saveOtaPackageData(@ApiParam(value = OTA_PACKAGE_ID_PARAM_
                                              @RequestPart MultipartFile file) throws ThingsboardException {
         checkParameter(OTA_PACKAGE_ID, strOtaPackageId);
         checkParameter(CHECKSUM_ALGORITHM, checksumAlgorithmStr);
+        OtaPackageId otaPackageId = new OtaPackageId(toUUID(strOtaPackageId));
+        OtaPackageInfo otaPackageInfo = checkOtaPackageInfoId(otaPackageId, Operation.READ);
         try {
-            OtaPackageId otaPackageId = new OtaPackageId(toUUID(strOtaPackageId));
-            OtaPackageInfo info = checkOtaPackageInfoId(otaPackageId, Operation.READ);
 
-            OtaPackage otaPackage = new OtaPackage(otaPackageId);
-            otaPackage.setCreatedTime(info.getCreatedTime());
-            otaPackage.setTenantId(getTenantId());
-            otaPackage.setDeviceProfileId(info.getDeviceProfileId());
-            otaPackage.setType(info.getType());
-            otaPackage.setTitle(info.getTitle());
-            otaPackage.setVersion(info.getVersion());
-            otaPackage.setTag(info.getTag());
-            otaPackage.setAdditionalInfo(info.getAdditionalInfo());
 
             ChecksumAlgorithm checksumAlgorithm = ChecksumAlgorithm.valueOf(checksumAlgorithmStr.toUpperCase());
-
-            byte[] bytes = file.getBytes();
+            byte[] data = file.getBytes();
             if (StringUtils.isEmpty(checksum)) {
-                checksum = otaPackageService.generateChecksum(checksumAlgorithm, ByteBuffer.wrap(bytes));
+                checksum = otaPackageService.generateChecksum(checksumAlgorithm, ByteBuffer.wrap(data));
             }
-
-            otaPackage.setChecksumAlgorithm(checksumAlgorithm);
-            otaPackage.setChecksum(checksum);
-            otaPackage.setFileName(file.getOriginalFilename());
-            otaPackage.setContentType(file.getContentType());
-            otaPackage.setData(ByteBuffer.wrap(bytes));
-            otaPackage.setDataSize((long) bytes.length);
-            OtaPackageInfo savedOtaPackage = otaPackageService.saveOtaPackage(otaPackage);
-            logEntityAction(savedOtaPackage.getId(), savedOtaPackage, null, ActionType.UPDATED, null);
-            return savedOtaPackage;
+            return tbOtaPackageService.saveOtaPackageData(otaPackageInfo, checksum, checksumAlgorithm,
+            data, file.getOriginalFilename(), file.getContentType(), getCurrentUser(), null);
         } catch (Exception e) {
-            logEntityAction(emptyId(EntityType.OTA_PACKAGE), null, null, ActionType.UPDATED, e, strOtaPackageId);
+            tbOtaPackageService.saveOtaPackageData(otaPackageInfo, null, null,
```

Comment:
remove this, you should catch exceptions in the service

Reply:
Ok